src/BinaryParsers/PEBinary/PortableExecutable/PE.cs

```diff
@@ -527,7 +527,7 @@ public bool IsManagedResourceOnly
                     return this.isManagedResourceOnly.Value;
                 }
 
-                this.isManagedResourceOnly = this.metadataReader.MethodDefinitions.Count == 0;
+                this.isManagedResourceOnly = this.metadataReader?.MethodDefinitions.Count == 0;

```

Comment:
maybe you can just have a check for IsManaged at the top of this helper? would be clearer, I think.

if (!this.IsManaged) { return false;}

if you do that, afterwards you could Debug.Assert on this.metadataReader being non-null.